nilearn/maskers/_mixin.py

```diff
@@ -431,16 +431,8 @@ def generate_report(self, title=None) -> list[None] | HTMLReport:
 
         return generate_report(self)
 
-    def _reporting(self):
-        # if report is disabled or the model is not yet fitted
-        if self.reports is False or self.__sklearn_is_fitted__() is False:
-            self._report_content["summary"] = None
-            return [None]
```

Comment:
These checks were comment in all `masker._reporting`, so I had decided to put them there. However, I've realized that this is actually dead code. If these conditions are satisfied, an empty report with warnings is returned earlier.

For the same reason I've removed the check in estimator checks.

This also satisfies the requirement [here](https://github.com/nilearn/nilearn/issues/5828#issue-3596683340).